// json-schema-to-typescript/dist/test/e2e/deep.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.input = {
    type: 'object',
    properties: {
        foo: {
            type: 'object',
            oneOf: [
                {
                    oneOf: [
                        { type: 'number' },
                        { $ref: '#/definitions/foo' },
                        { $ref: '#/definitions/bar' },
                        {
                            properties: {
                                baz: { type: 'number' }
                            }
                        }
                    ]
                },
                { $ref: '#/definitions/bar' }
            ]
        }
    },
    definitions: {
        foo: {
            properties: {
                a: { type: 'string' },
                b: { type: 'integer' }
            },
            additionalProperties: false,
            required: ['a', 'b']
        },
        bar: {
            properties: {
                a: { type: 'string' }
            },
            required: ['a', 'b']
        }
    },
    required: ['foo'],
    additionalProperties: false
};
exports.output = "export interface Deep {\n  foo: ((number | Foo | Bar | {\n    baz?: number;\n    [k: string]: any;\n  }) | Bar);\n}\nexport interface Foo {\n  a: string;\n  b: number;\n}\nexport interface Bar {\n  a: string;\n  [k: string]: any;\n}\n";
